add lua support to full install & merge old nginx configuration

vdloo opened the pull request:
- strip some unused stuff like [here](https://github.com/ByteInternet/nginx-hypernode/pull/1/files)
- add nginx-lua to nginx-full
- statically build `--with-http_geoip_module=` because loading it dynamically as .so doesn't work for some reason and it doesn't really matter for us (we want the dynamic module loading future for adding modules without recompiling, not for removing them)

Commit:
add lua support to full install

gwillem commented:
nice! does it upgrade without hiccup? probably we should include our own upstart script?

vdloo committed:
add Hypernode nginx upstart script

- infinite respawn
- daemon off so we don't wait for a fork (in case of occupied sockets) see [this](https://www.byte.nl/blog/upstart-respawn-is-not-respawning)
- slay dangling children if they are occupying the socket

vdloo commented:
Yeah we can remove the upstart script from the hypernode-control playbooks once all nodes have this build deployed
